types/networks/src/lib.rs

```diff
@@ -131,6 +106,11 @@ pub struct HeightInfo {
     pub epoch: ChainEpoch,
 }
 
+pub fn sort_by_epoch(mut height_info_vec: Vec<HeightInfo>) -> Vec<HeightInfo> {
+    height_info_vec.sort_by(|a, b| a.epoch.cmp(&b.epoch));
+    height_info_vec
+}
+
```

Comment:
If this function takes a slice of `HeightInfo` (and turns it into a Vec) then the rest of the code will become simpler.